zebra-consensus/src/chain.rs

```diff
@@ -118,11 +137,13 @@ where
             .expect("hardcoded checkpoint list extends past sapling activation")
     };
 
+    // Correctness:
+    //
+    // We use `ServiceExt::oneshot` to make sure every `poll_ready` has a
+    // matching `call`. See #1593 for details.
     let tip = match state_service
-        .ready_and()
-        .await
-        .unwrap()
-        .call(zs::Request::Tip)
+        .clone()
+        .oneshot(zs::Request::Tip)
         .await
         .unwrap()
```

Comment:
This fix is part of #1675 - and it's actually something I missed during my review last week.

Since we're fixing this file, let's also use `ServiceExt::oneshot` here.